scio-core/src/main/scala/com/spotify/scio/values/SCollection.scala

```diff
@@ -308,15 +309,15 @@ sealed trait SCollection[T] extends PCollectionWrapper[T] {
    * this SCollection, and then flattening the results.
    * @group transform
    */
-  def flatMap[U: ClassTag](f: T => TraversableOnce[U]): SCollection[U] =
+  def flatMap[U: ClassTag](f: T => GenTraversableOnce[U]): SCollection[U] =
     this.parDo(Functions.flatMapFn(f))
 
   /**
    * Return a new SCollection[U] by flattening each element of an SCollection[Traversable[U]].
    * @group transform
    */
-  def flatten[U: ClassTag](implicit ev: T <:< TraversableOnce[U]): SCollection[U] =
-    flatMap(_.asInstanceOf[TraversableOnce[U]])
+  def flatten[U: ClassTag](implicit ev: T => GenTraversableOnce[U]): SCollection[U] =
+    flatMap(e => e)
```

Comment:
should be `.flatMap(ev)`?

Reply:
@nevillelyh 🤦‍♂️ yes!

if we decide to keep `TraversableOnce` than this can be `flatMap(e => ev(e).seq)`

Reply:
Do we need `.seq` in this case? It returns a sequential version, I guess to preserve order. But `SCollection` doesn't guarantee order anyway.